### PR TITLE
Feature/2905 debug source code by label

### PR DESCRIPTION
**Pull request by tmandys**

### CODE REVIEW

File: doxygen/lang/900_release_notes.dox.tmpl [hunk 1]
```diff
@@ -7,6 +7,14 @@
     @par Release Summary
     Bugfix release; see details below
 
+    @subsection qore_08137_new_features New Features in Qore
+    - improved debugging support:
+        - @ref Qore::ProgramControl::getStatementIdInfo() "ProgramControl::getStatementIdInfo()" provides breakpoint info
+        - command to resolve statement
+        - interrupt notification provides call stack info
+        - "Run to statement" implementation
+        - @ref DebugHandler reimplemented to support multiple websocket handlers
```
Comment (davidnich) [inline doxygen/lang/900_release_notes.dox.tmpl]: `@ref DebugHandler` will not work; you have to use `<a href="../../modules/DebugHandler/html/index.html">DebugHandler</a>` instead.  Please always generate the documentation and review the output before submitting a pull request

File: examples/test/qore/debug/test-debug.qtest [hunk 2]
```diff
@@ -70,17 +70,18 @@ class MyDebugProgram inherits DebugProgram {
...
-  handle(int func, ProgramControl pgm, reference runState, *int statementId, *hash extra) {
+  handle(int func, ProgramControl pgm, reference runState, reference runToStatementId, *int statementId, *hash extra) {
     *hash location;
     handleCount.inc();
     if (statementId)
       location = pgm.getStatementIdInfo(statementId);
-    #printf("%y: pgm: %y, %y, %y, %y\n", func, pgm, runState, location, extra);
+    #printf("%y: pgm: %y, %y, %y, %y, %y\n", func, pgm, runState, runToStatementId, location, extra);
     hash it = (
       'func': stResolve{func},
```
Comment (davidnich) [inline examples/test/qore/debug/test-debug.qtest]: does this file have 2-space indentation?  it should be 4

File: examples/test/qore/debug/test-debug.qtest [hunk 3]
```diff
@@ -70,17 +70,18 @@ class MyDebugProgram inherits DebugProgram {
...
       'file': location.file,
       'line': location.start_line,
       'tid': gettid(),
+      'runToStatementId': runToStatementId,
     );
     any act = getAction(func);
```
Comment (davidnich) [inline examples/test/qore/debug/test-debug.qtest]: please use `auto` instead of `any` as `any` is subject to type stripping (performance issue)

**Review by davidnich — CHANGES_REQUESTED** [review]

**Comment by tmandys** [comment]
requested changes from review done

**Comment by davidnich** [comment]
@tmandys as per my email this weekend, I would like to retarget this and all other enhancements to libqore to develop, let's discuss if you have any objections

**Pull request closed without merging**